raven/src/main/java/com/getsentry/raven/connection/RandomEventSampler.java

```diff
@@ -46,6 +44,7 @@ public RandomEventSampler(double sampleRate, Random random) {
      */
     @Override
     public boolean shouldSendEvent(Event event) {
-        return Math.abs(random.nextInt()) % RATE_MULTIPLIER < sampleRate;
+        double randomDouble = random.nextDouble();
+        return sampleRate > Math.abs(randomDouble);
```

Comment:
Probably OK to just make this `>=` for extra safety in case where it says "approximately" in [the documentation on `nextDouble`](https://docs.oracle.com/javase/7/docs/api/java/util/Random.html#nextDouble()) means "approximately not actually exclusive just kidding."